File: Distancia2puntosC3d/Distancia2puntosC/Punto_X.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Distancia2puntosC
{
    class Punto_X
    {
        protected double  x;
        public double X
        {
            get{ return x; }
            set { x = value; }
        }

        public Punto_X()
        {
            x = 0;
            
        }
        public Punto_X(double X)
        {
            this.x  = X;
          
        }

     

            public double CalcularDistancia(Punto_X a)
         {
           return Math.Abs(x - a.X);
       }
        public override string ToString()
        {
        return x + " ";
        }
        }
    }

File: Distancia2PuntosVB3d/Distancia2PuntosVB/PuntoX.vb
Public Class PuntoX
    Protected x As Double

    Public Property _X As Double
        Get
            Return x
        End Get
        Set(ByVal value As Double)
            x = value
        End Set
    End Property

    Public Sub New()
        x = 0
    End Sub

    Public Sub New(ByVal _X As Double)
        Me.x = _X
    End Sub

    Public Function CalcularDistancia(ByVal a As PuntoX) As Double
        Return Math.Abs(x - a._X)
    End Function

    Public Overrides Function ToString() As String
        Return x & " "
    End Function
End Class
